File: app/src/main/java/com/example/seyahatkolaymobileapp/splash_screen.java
package com.example.seyahatkolaymobileapp;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class splash_screen extends AppCompatActivity {
    private static final long SPLASH_TIME_OUT = 1000; // Splash screen görüntüleme süresi (milisaniye cinsinden)

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash_screen);
        EdgeToEdge.enable(this);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                Intent I1 = new Intent(splash_screen.this, Login.class); // Ana aktivitenin Intent'i
                startActivity(I1);
                finish(); // Splash ekranını kapat
            }
        }, SPLASH_TIME_OUT);
    }

}

File: app/src/main/java/com/example/seyahatkolaymobileapp/splash_screen.kt
package com.example.seyahatkolaymobileapp

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.view.View
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class splash_screen : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash_screen)
        this.enableEdgeToEdge()
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v: View, insets: WindowInsetsCompat ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        Handler().postDelayed({
            val I1 = Intent(this@splash_screen, Login::class.java) // Ana aktivitenin Intent'i
            startActivity(I1)
            finish() // Splash ekranını kapat
        }, SPLASH_TIME_OUT)
    }

    companion object {
        private const val SPLASH_TIME_OUT: Long =
            1000 // Splash screen görüntüleme süresi (milisaniye cinsinden)
    }
}

